text/XXX-A-Compilation-of-Short-Interactive-Tutorials-and-How-To-Guides-for-Coq.md

```diff
@@ -0,0 +1,257 @@
+- Title: A Compilation of Short Interactive Tutorials and How-To Guides for Coq
+- Drivers:
+    - Thomas Lamiaux / @thomas-lamiaux / thomas.lamiaux@ens-paris-saclay.fr
+    - Theo Zimmermann / @Zimmi48 / theo.zimmermann@telecom-paris.fr
+    - Pierre Rousselin / @Villetaneuse / rousselin@math.univ-paris13.fr
+
+----
+## Table of Content
+
+1. Summary
+2. Motivation
+   - 2.1 The lack of action-oriented documentation
+   - 2.2 Why it is important
+3. A compilation of interactive tutorials and how-to guides
+   - 3.1 General Design
+   - 3.2 A Horizontal Documentation
+   - 3.3 Online interface
+4. Organisation
+   - 4.1 First Steps
+   - 4.2 Writing the documentation
+   - 4.3 Maintenance and evolution
+   - 4.4 Web Interface
+   - 4.5 Moving on to the official Coq repository and Coq's webpage
+   - 4.6 Human resources
+
+
+---
+## 1. Summary
+
+The goal of this CEP is to discuss the design of an online compilation of short and interactive tutorials and how-to guides for Coq and the Coq Platform.
+
+Each core functionality and plugin of Coq and the Coq Platform should have (short) pedagogical tutorials and/or how-to guides demonstrating how to use the functionality, with practical examples. Tutorials and how-to guides serve different purposes and are complementary: tutorials guide a user to discover a specific feature like "Notations in Coq", by going through predetermined examples, and introducing notions gradually, whereas how-to guides are use-case-oriented and use examples to show what to do in front of a specific problem (and possibly the various alternatives to address it). See https://diataxis.fr/ for a reference definition of tutorial vs how-to guide (in particular, https://diataxis.fr/tutorials-how-to/ for a comparison).
+
+Compared to the reference manual, such a documentation is not meant to be complete or going into low level details. It is meant to be pedagogical and action-oriented, so that it enables users to learn how to use new features autonomously and horizontally : these tutorials and how-to guides should be as self-contained as possible to have only a short (if any) list of prerequisites. Moreover, by being horizontal, such a documentation could provide different tutorials (for some specific features) to cover the needs of users with different backgrounds.
```

Comment:
> Some tutorials should be read before others.

Off course, and it is the case for Equations. Simply, you should not hesitate to recall a concept in 20 lines if it avoid having to read a 1000 lines tutorial.

Typically, I have read papers with bad practices, e.g. "it is not possible cf [32]" so you spend 3h looking for it, and in the end the answer is one line, so then please write "it is not possible because constructors do not preserve variables [32]" .

That is basically the idea.